### DISCUSSION TITLE
Container image size

### DISCUSSION THREAD

**Discussion by ralf-berger** [ude-soco/CourseMapper-webserver#1050] [General]
@ude-soco/developers The [coursemapper-webserver-coursemapper-kg-concept-map](https://hub.docker.com/layers/socialcomputing/coursemapper-webserver-coursemapper-kg-concept-map/edge/images/sha256-1fdfb969bdca43f22939d56bc72c81f58e445257b328bdba9e4a2480b5768a26?context=repo) is huge (4.23 GB compressed) and takes a lot of time to build, deploy etc. I think it's because of image sizes we're also using up our free GitHub Actions compute time much faster than we used to. E.g. the aforementioned takes about 9 minutes to build, without any testing.

**Comment by jeanqussa** [comment]
Is this blocking the current deployment on Edge?

**Comment by rawaa123** [comment]
@Ralf-Berger,
We are looking to increase the compute time for GitHub Actions, as the current changes are not being reflected in the deployment. Could you please provide me with the best options and pricing details. 
We need to proceed with the purchase as soon as possible because we urgently need to deploy it so we have enough time to test and fix the bugs before the start of the next semester. Looking forward to your understanding and cooperation.

**Reply by ralf-berger** [reply]
There is a GitHub Teams plan which includes a moderate increase in build minutes, but is billed per collaborator and would cost a fortune (as in 1500+ USD per year). Other than that, the rate per build minute using regular runners is 0.008 USD (the 2000 free minutes that we ran through pretty quickly this month would be equivalent to 16 USD). To use paid build time, you need to add a means of payment to the account and set a spending limit.

Generally speaking, it's a good idea to make as many projects public as possible, since [public projects do not consume any of the free compute minutes](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#about-billing-for-github-actions). Turning any private project public should resume its builds, even without free minutes left.

For private projects in particular we should try to keep our build-times low by cleaning up dependencies, keeping data/models out of containers, using production settings everywhere, using build caches, etc.

**Reply by rawaa123** [reply]
Do you mean if we make our project public then will rebuild even thoug if we are running out of minutes?

**Reply by ralf-berger** [reply]
Yes, that's my understanding.

**Reply by rawaa123** [reply]
We have agreed to make it public and rebuild it. Please once this done inform us so we ca start testing and fxing the bugs. your help is highly appreciated.

**Reply by rawaa123** [reply]
The image has been successfully rebuilt , so we can start testing on Edge. Thanks a lot for your help.

**Reply by ralf-berger** [reply]
Seems to be updating, just need to figure out what's going on with #1058

**Reply by rawaa123** [reply]
The issue arises from the old service for creating the KG still running, which is preventing the new service in "coursemapper-kg" from being invoked. 

It is worth mentioning, we’ve updated the file and directory structure in 'coursemapper-kg'and now we have new structure, so I believe we need to rebuild the project from scratch and pull the latest changes to reflect the new hierarchy. or another suggestion??

**Reply by ralf-berger** [reply]
Please check everything is running as expected now.

**Reply by rawaa123** [reply]
We got this error msg and i think it is server side. This probably means the server terminated abnormally before or while processing the request. any suggestions? your help is highly appreciated

![image](https://github.com/user-attachments/assets/91fca6ee-9d25-4d9b-a1b3-a4decb79f8f8)

**Reply by rawaa123** [reply]
@ralf-berger any help or suggestion fro the previos issue. Till now we are unable to test this part of this ystem and solve bugs. Looking forward to your understanding and cooperation.

**Reply by ralf-berger** [reply]
@rawaa123 Please re-open #1075 if the error reappeared.
